### PR TITLE
Feature/network switcher

### PR DESCRIPTION
**Pull request by alexcherman**

### CODE REVIEW

**Comment by zamli** [comment]
image
dropdowns should open up at the same place where settings is.

**Comment by zamli** [comment]
Same thing for this drop down. 
image

**Comment by zamli** [comment]
image
clicking change wallet should open up wallet connector again
image

**Comment by zamli** [comment]
image
settings click area right now is only the cogwheel. Make the bg for the settings open up settings too

**Comment by alexcherman** [comment]
> image
> clicking change wallet should open up wallet connector again
> image

@luzzif How I can disconnect account programatically without metamask?

**Comment by zamli** [comment]
Open network switcher popup/dropdown at the same position as the wallet connect popup
image
image

**Comment by zamli** [comment]
image

Add this popup when going from x network to ethereum. There is no switching to ethereum built in to metamask. So we need to ask our users to switch manually from metamask.

https://www.figma.com/file/L8tamoiv3mTCRaypHLk1us/DXswap?node-id=3641%3A128

**Comment by zamli** [comment]
Noticed the menu opens above the token preview.
image

**Comment by zamli** [comment]
image
Let's keep padding at 1 rem for mobile too

**Comment by zamli** [comment]
image
Make the bg for the settings button 0 Alpha. Area should still be clickable.

**Comment by zamli** [comment]
image
on mobile:

- Only show first 3 numbers after decimal on the Ethereum Balance. Example 1.021.
- Show only first 4 letters/numbers for wallet and 4 ending. Example 0XB1...315F

**Comment by zamli** [comment]
Both desktop and mobile:
image
Show wallet address in all caps/uppercase

**Comment by alexcherman** [comment]
What do you mean by alpha 0? Right now there is solid background(#191A24) without opacity and the whole area is clickable.

**Comment by alexcherman** [comment]
> image
> on mobile:
> 
> * Only show first 3 numbers after decimal on the Ethereum Balance. Example 1.021.
> * Show only first 4 letters/numbers for wallet and 4 ending. Example 0XB1...315F

What about desktop?

**Comment by zamli** [comment]
> What do you mean by alpha 0? Right now there is solid background(#191A24) without opacity and the whole area is clickable.

what i mean is making the background transparent. There is no color.
image

**Comment by zamli** [comment]
> > image
> > on mobile:
> > 
> > * Only show first 3 numbers after decimal on the Ethereum Balance. Example 1.021.
> > * Show only first 4 letters/numbers for wallet and 4 ending. Example 0XB1...315F
> 
> What about desktop?

we can keep it on desktop

File: src/components/Header/index.tsx [hunk 1]
```diff
@@ -248,38 +265,40 @@ function Header({ history }: { history: any }) {
...
-            ) : null}
-            <Web3Status />
-          </AccountElement>
+          <Web3Status />
+          <DesktopSettingsWrapper>
+            <Settings />
+          </DesktopSettingsWrapper>
         </HeaderElement>
-        {!isMobile && <Settings />}
+        <HeaderSubRow>
```
Comment (luzzif) [inline src/components/Header/index.tsx:289]: This needs to be commented out

**Review by luzzif — APPROVED** [review]

File: src/components/Header/index.tsx (hunk 1 above)
Comment (alexcherman) [reply src/components/Header/index.tsx:289]: Sorry, could you explain it in more details?

**Comment by luzzif** [comment]
This is the screen I see on mobile as soon as I open the dapp without a connected wallet:
image

Shouldn't we show a dropdown somewhere to change the currently connected network (like we currently do) WITHOUT "forcing" the user to connect the wallet? Maybe they just want to check out prices on different networks.

**Comment by luzzif** [comment]
Clicking on the "connect wallet" button above opens an empty choice dropdown again. I think we briefly discussed this on Keybase.
image

**Comment by luzzif** [comment]
Do we want outlines on buttons @zamli?

**Comment by zamli** [comment]
> Do we want outlines on buttons @zamli?
No, there should be no outlines as far as i designed it.

**Comment by zamli** [comment]
Added a network switcher for the not connected mode. This is something I totally missed while designing this.

On mobile
not connected

Desktop
image

**Comment by alexcherman** [comment]
> > Do we want outlines on buttons @zamli?
> 
> No, there should be no outlines as far as i designed it.

Are you sure that we should remove outlines before we add our owns focus styles? In that case user won't be able to navigate using keyboard.

**Comment by alexcherman** [comment]
However, if I remove default focus styles, it will be easier for me find buttons where we need to add new focus styles. Ok, I'll do it.

**Comment by zamli** [comment]
Metamask mobile is not automatically switching or adding networks.
Please see how https://chainlist.org/ is doing it

**Comment by 0xVenky** [comment]
Noticed this small misalignment in this build, while its working perfectly in swapr.eth
image

In Swapr.eth

image

**Comment by zamli** [comment]
image
Give this loader spacing 5px on left side.

**Comment by zamli** [comment]
Give settings button a BG like the rest
image

How it should look like
image

**Comment by zamli** [comment]
image
when eth balance is 0, make the token preview for it red.
On xdai: when xdai is 0, make preview red

**Pull request merged**